src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -205,6 +328,46 @@ IN_PROCESS_APPLICATION::~IN_PROCESS_APPLICATION()
     Recycle();
 }
 
+REQUEST_NOTIFICATION_STATUS
+IN_PROCESS_APPLICATION::OnAsyncCompletion(
+    IHttpContext*           pHttpContext,
+    DWORD                   cbCompletion,
+    HRESULT                 hrCompletionStatus
+)
+{
+    HRESULT hr;
+    IN_PROCESS_STORED_CONTEXT* pInProcessStoredContext = NULL;
+
+    hr = GetInProcessStoredContext(pHttpContext, &pInProcessStoredContext);
+    if (FAILED(hr))
+    {
+        // Finish the request as we couldn't get the callback
+        return RQ_NOTIFICATION_FINISH_REQUEST;
+    }
+    // Call the managed handler for async completion.
+    return m_ManagedContextHandler(pInProcessStoredContext->QueryManagedHttpContext(), hrCompletionStatus, cbCompletion);
+}
+
+HRESULT
+IN_PROCESS_APPLICATION::GetInProcessStoredContext(
```

Comment:
This method has no dependency on any other members of IN_PROCESS_APPLICATION. You can make it as a standalone method or exported on